server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/Redirect.java

```diff
@@ -281,142 +285,121 @@
-        String addressList = getInitParameter("to", getInitParameter("recipients"));
-
-        // if nothing was specified, return null meaning no change
-        if (addressList == null) {
+        String toOrRecipients = getToOrRecipients();
+        if (toOrRecipients == null) {
             return null;
         }
-
-        try {
-            iaarray = InternetAddress.parse(addressList, false);
-            for (int i = 0; i < iaarray.length; ++i) {
-                String addressString = iaarray[i].getAddress();
-                MailAddress specialAddress = getSpecialAddress(addressString, new String[] { "postmaster", "sender", "from", "replyTo", "reversePath", "unaltered", "recipients", "to", "null" });
-                if (specialAddress != null) {
-                    iaarray[i] = specialAddress.toInternetAddress();
-                }
-            }
-        } catch (Exception e) {
-            throw new MessagingException("Exception thrown in getTo() parsing: " + addressList, e);
+        if (toOrRecipients.isEmpty()) {
+            throw new MessagingException("Failed to initialize \"recipients\" list; empty <recipients> init parameter found.");
         }
-        if (iaarray.length == 0) {
-            throw new MessagingException("Failed to initialize \"to\" list; empty <to> init parameter found.");
+        List<InternetAddress> list = Lists.newArrayList();
+        for (InternetAddress address : extractAddresses(toOrRecipients)) {
+            list.add(toMailAddress(address).toInternetAddress());
         }
+        return list.toArray(new InternetAddress[list.size()]);
+    }
 
-        return iaarray;
+    private String getToOrRecipients() throws MessagingException {
+        return getInitParameter("to", getInitParameter("recipients"));
     }
 
-    /**
-     * @return the <code>reversePath</code> init parameter or the postmaster
-     *         address or <code>SpecialAddress.SENDER</code> or
-     *         <code>SpecialAddress.NULL</code> or <code>null</code> if missing
-     */
+    @Override
     protected MailAddress getReversePath() throws MessagingException {
         String addressString = getInitParameter("reversePath");
-        if (addressString != null) {
-            MailAddress specialAddress = getSpecialAddress(addressString, new String[] { "postmaster", "sender", "null" });
-            if (specialAddress != null) {
-                return specialAddress;
-            }
+        if (addressString == null) {
+            return null;
+        }
 
-            try {
-                return new MailAddress(addressString);
-            } catch (Exception e) {
-                throw new MessagingException("Exception thrown in getReversePath() parsing: " + addressString, e);
-            }
+        MailAddress specialAddress = getSpecialAddress(addressString, new String[] { "postmaster", "sender", "null" });
+        if (specialAddress != null) {
+            return specialAddress;
```

Comment:
Can't you modify getSpecialAddress to take a default value as last parameter? You use this construct twice.